src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/AuthorisedGetCriteriaOperation.java

```diff
@@ -63,15 +63,21 @@ private boolean criteriaAllowedByCRUD(CaseType caseType, CriteriaInput criteriaI
     }
 
     private boolean filterDistinctFieldsByRole(final Set<String> addedFields, final CriteriaInput criteriaInput, final Set<String> userRoles) {
-        if (addedFields.contains(criteriaInput.getField().getId())) {
+        String id = buildFieldId(criteriaInput);
+        if (addedFields.contains(id)) {
             return false;
         } else {
             if (StringUtils.isEmpty(criteriaInput.getRole()) || userRoles.contains(criteriaInput.getRole())) {
-                addedFields.add(criteriaInput.getField().getId());
+                addedFields.add(id);
                 return true;
             } else {
                 return false;
             }
         }
     }
+
+    private String buildFieldId(CriteriaInput criteriaInput) {
+        return StringUtils.isEmpty(criteriaInput.getField().getElementPath()) ? criteriaInput.getField().getId() :
+            criteriaInput.getField().getId() + '.' + criteriaInput.getField().getElementPath();
+    }
```

Comment:
consider moving this to a method in the `CriteriaInput` class itself for future reuse in other places as well. For example `getPath` or `getFullId`.